### ISSUE TITLE
Predcit function result is different from Training-Val using the same data

### ISSUE THREAD

**Issue by Marvinmw** [tensorflow/tensorflow#40478]
Version:
Tensorflow 2.0.0 
Keras 2.3.1
Ubuntu 18

I use predict to check the model performance on val data but it is totally different from the training log file. **Predict and Val in the training use the same dataset.**

The top layers of my model:
`        
        model = Sequential()
        model.add(Embedding(input_dim = num_words,
                            output_dim=EMBEDDING_DIM,
                                    weights=[emb],  # remove this if you want to train your weight
                                    input_length=maxlen,
                                    trainable=train_embedding,
                                    mask_zero=True))
        ........
        .........
         model.add(Dropout(0.2))
        # Heuristic model: outputs 1-of-num_classes prediction
        model.add(BatchNormalization())

        model.add(Dense(dense_layer_size, activation="relu"))

        model.add(Dense(1, activation="sigmoid"))

        self.model = model

        self.model.compile(
            optimizer="adam",
            loss=tf.keras.losses.BinaryCrossentropy(from_logits=True),
            metrics=[
                  tf.keras.metrics.AUC(curve="PR", name="PR"),
                      tf.keras.metrics.AUC(curve="ROC", name="ROC"),
                    "accuracy",
                    tf.keras.metrics.Precision(),tf.keras.metrics.Recall(),
                     tf.keras.metrics.binary_accuracy],
        )
`

Train code
`
self.model.fit_generator(train_generator, epochs=epochs, verbose=verbose,
                                     validation_data = validation_generator,
                                     shuffle=True, callbacks=[checkpoint], class_weight=class_weights)
`
Prediction and Compute AUC-PR code:

`

padded_x = keras.preprocessing.sequence.pad_sequences(x_test, maxlen=max_len)

classlable, pro = model.predict(padded_x, batch_size)
pro = np.squeeze(pro)
y_test = np.squeeze(y_test)
from sklearn import metrics
precision, recall, thresholds = metrics.precision_recall_curve(y_test, pro)
pr = metrics.auc(recall, precision)
print("PR {}".format(pr))

`
Use Prediction compute PR-AUC:  0.2803784709997255
But  Val_PR: 0.4

I tried some solutions I found, use model.save.save or from keras.backend import manual_variable_initialization
manual_variable_initialization(True).
1. When I save the whole model weights, does the save function also save the Embedding layer?
2. I am doing binary-classification task, am I correctly using the metrics in model.compile  function?

**Comment by Marvinmw** [comment]
I am try using fit function instead of fit_generator.

**Comment by Marvinmw** [comment]
Hi, I use tensorflow.keras instead of keras and upgrade tensorflow to 2.2.0. Then it is fine.

**Comment by Saduf2019** [comment]
@Marvinmw 
Does that resolve the problem, if it does please move this issue to closed status.

**Comment by Marvinmw** [comment]
yes.solved.
